includes/class-send-list.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Newspack Newsletters Send List
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Newsletters;
+
+use Newspack_Newsletters;
+use WP_Post;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class used to represent one Send_List.
+ *
+ * A Send_List can be either a top-level list or a sublist.
+ * A sublist must specify a parent list ID.
+ */
+class Send_List {
+	/**
+	 * The configuration associated with this Send_List.
+	 *
+	 * @var array
+	 */
+	protected $config;
+
+	/**
+	 * Initializes a new Send_List.
+	 *
+	 * @param array $config The configuration for the Send_List.
+	 * @throws \InvalidArgumentException In case the Send_List object can't be built from the given config data.
+	 */
+	public function __construct( $config ) {
+		$schema = self::get_config_schema();
+		$errors = [];
```

Comment:
NIT suggestion: the `WP_Error` class provides a nice API with `$errors->add( $code, $message )` and `$errors->has_errors()`. I find it better than handling this with try/catch.

Reply:
Good idea. 7f4906e refactors the error handling to use a `WP_Error` object for this instead of an array of strings. The constructor will no longer throw an exception, but will add error data that can be fetched all at once using the new `get_errors()` method. `get_*` and `set_*` methods will also return any applicable errors in addition to adding their own messages to the class instance's errors object.

Reply:
#1631 will need to be refactored to take advantage of these improvements, but I'll hold off on that until this PR is merged.

Reply:
I failed to recognize that this is the constructor 🤦 

I think we can still leverage the WP_Error class to manage multiple errors but it's best practice to maintain the throw here... Tough decision because I really like how you did it 😅

Reply:
Ah, no problem—7a3d8b3 restores the throw but keeps the use of `WP_Error` to manage the errors.